### ISSUE TITLE
Self-signed cert on domain?

### ISSUE THREAD

**Issue by scouten-adobe** [creator-assertions/identity-assertion#101]
I'd suggest allowing a lightweight approach where the owner of a web domain (or fixed path) could create a self-signed cert, make its public key available at (something like) `<web domain>/.well-known/c2pa.jwks`, allowing anyone to fetch the key at validation time and make their own trust decision based on the web domain itself. This would allow scaling C2PA to entities (individuals and organizations) that are not part of an organized trust framework to self-sufficiently create identity assertions. A validator would present status message such as "this asset has been signed by `example.com`".

_Originally posted by @christianpaquin in https://github.com/creator-assertions/identity-assertion/issues/22#issuecomment-1960025145_

**Comment by puhley** [comment]
A challenge with this approach is what happens when you rotate the certificate?  You replace the public certificate on your site with a new certificate. Now all of your old content no longer validates because there is a different value there. You can append multiple certificates. However, you also have no revocation solutions if the key ever gets compromised. You can only remove the previous public certificate and invalidate all of the previous content.

**Comment by christianpaquin** [comment]
There is some leeway here, depending on what the security goals are. Keeping a public file with all your current and past expired certs would solve most of the issues. Self-revoked certs could indeed be deleted, invalidating all the content signed by them, as expected. If you want something more granular, you can add a revocation date in the cert file. One practical mechanism would be to use, e.g., a [JSON Web Key set](https://www.rfc-editor.org/rfc/rfc7517#section-5) stored at a `.well-known` location.

Bottom line, yes there are edge cases, but having a generic solution for "signing with your website" would be a plus, IMHO. The threat model for validating such assets would be ~equivalent to accessing them directly from the website (i.e., one is TLS asserted, the other one is asserted by a file asserted by TLS).
